tests/pos/i10693.scala

```diff
@@ -0,0 +1,8 @@
+def test[A, B](a: A, b: B): A | B = a
+val v0 = test("string", 1)
+val v1 = test(1, "string")
+val v2 = test(v0, v1)
+val v3 = test(v1, v0)
+val v4 = test(v2, v3)
+val v5 = test(v3, v2)
+val v6 = test(v4, v5)
```

Comment:
I'm glad the issue is being resolved, but I don't understand this test.

How can we see that for instance the type of `v2` isn't `String | Int | Int | String`? Don't we want to show that it is inferred as `Int | String` or `String | Int`?

Reply:
Right now: manual inspection after -Xprint:typer. It would be good if someone could add a better test.

Reply:
Now this bit I had figured out!
Its a bit of a kludge, but In #20014 I had a test that was actually validating this via the repl tests.

Reply:
Great! That's a good idea to use repl tests for that. I added your test to the PR.